15 cm 2 course management dashboard

flying-cmd opened the pull request:
Instructor clicks the Course button on the left to enter the Course Management interface, where users can View Review, edit, publish and delete courses.
![image](https://github.com/user-attachments/assets/bcfd05d1-d85b-4b60-bbfd-d435ecae4f5a)
![image](https://github.com/user-attachments/assets/cfd34206-8a03-4eaa-bf44-9bd2faf27b9f)
![image](https://github.com/user-attachments/assets/4a6d1ecb-08a8-4302-adf0-60d5b14f40c5)

RuntaoZhuge commented:
Can they have both list view and card view for the courses?

flying-cmd commented:
Here are why I think the above suggestions are unnecessary:
1. I don't think the error handling part code is difficult to read.
2. I think the `fetchCourses, updateCourse` logic is not complicated and easy to understand. I don't think it's necessary to split it into smaller files.
3. PATCH and DELETE are operations for a single course, I don't think batching operations are necessary. The revenue calculation logic should be related to students purchasing courses, not related to the current course management dashboard. It is not implemented yet, I think it's better to open a new ticket for it.
4. I think these requirements are part of the middleware, maybe a new ticket can be opened for them.
5. I don't think there is any type issue because all the code can pass the eslint check.

RuntaoZhuge commented:
Please resolve the conflicts and also as instructors they should also be able to add a course on the panel I believe

flying-cmd commented:
After clicking the "New Course" button, the instructor will be navigated to the create new course form page to create a new course.
![image](https://github.com/user-attachments/assets/d93c993c-efdf-48c2-bc67-9f42715bba3a)

flying-cmd commented:
The following is the reason why I think above suggestions are unnecessary:
1. The logic in route.ts is not complicated. I have included necessary comments and I think it is easy to understand. I have decomposed the `ReviewsPageClient.tsx` component. The remaining parts are about fetching data. I don't think it is necessary to decompose them.
2. The main logic of `ReviewsPageClient.tsx` is fetching data. It is not complicated. I don't think it is necessary to decompose it further. The functions in `CourseRequest.tsx` are about sending requests to API endpoint, and more detailed error handling should be returned via the API endpoint (I have already implemented this), so I don't think it is necessary to have more detailed error handling in `CourseRequest.tsx`.
3. I think the dependency of the `ReviewsPageClient.tsx` rendering component is `reviews data`, and I have cached this data by tanstack react query, and I don't know what else needs to be cached.
4. I think rate limiting and brute force protection are part of the middleware, and I haven't handled them yet. Maybe open a new ticket for them is better. I have added a check in the `DELETE` and `PATCH` endpoints to determine whether the data belongs to the user, and I don't know why the AI ​​bot doesn't realize it.
5. `IReview` does not completely replace the interface to be used by the `ReviewsPageClient.tsx` component. I think the error handling in the route.ts file is consistent.
